with-formdata-image-upload/app/App.js

```diff
@@ -117,7 +115,17 @@ export default class App extends React.Component {
     alert('Copied image URL to clipboard');
   };
 
+  _askPermission = async (type, failureMessage) => {
+    const { status, permissions } = await Permissions.askAsync(type);
+
+    if (status === 'denied') {
+      alert(failureMessage);
+    }
+  };
+
   _takePhoto = async () => {
+    await this._askPermission(Permissions.CAMERA, 'We need the camera permission to take a picture...');
+    await this._askPermission(Permissions.CAMERA_ROLL, 'We need the camera-roll permission to read pictures from your phone...');
```

Comment:
just needed to add this for takePhoto to work (if you select it before you select pick image from camera roll)